src/Core/Validation/FieldValidation/SymfonyFieldValidatorTrait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use LogicException;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\ConstraintValidator;
+use SilverStripe\Core\Validation\FieldValidation\SymfonyFieldValidatorInterface;
+
+/**
+ * Trait for FieldValidators which validate using Symfony constraints
+ */
+trait SymfonyFieldValidatorTrait
+{
+    protected function validateValue(): ValidationResult
+    {
+        $result = parent::validateValue();
+        if (!$result->isValid()) {
+            return $result;
+        }
+        if (!is_a($this, SymfonyFieldValidatorInterface::class)) {
+            $message = 'Classes using SymfonyFieldValidatorTrait must implement SymfonyFieldValidatorInterface';
+            throw new LogicException($message);
+        }
```

Comment:
Put this before the parent validation - otherwise devs may miss it during development but it get hit in production.

Reply:
Updated